### PR TITLE
Ftc vesting contract

### PR DESCRIPTION
**Pull request by 0xModene**

### CODE REVIEW

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:77]: This type will break your tests

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:115]: You need to add an await here

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:135]: After you deploy your vesting contract, you must fund it with the correct amount of INDEX

File: test/token/FTCVesting.spec.ts
Comment (0xModene) [reply test/token/FTCVesting.spec.ts:77]: you mean the x? that was intentional (for now) to test things and single out the #claim test

Comment (ncitron) [reply test/token/FTCVesting.spec.ts:77]: Oh I see, I did not realize that it was intentional

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:27]: You can use array destructuring here to clean this up a bit like this:
```ts
[
    owner,
    treasury,
    recipient
] = await getAcounts();
```

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:111]: No need to have this expect here. This is covered by the prior test.

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:196]: Look into using `increaseTimeAsync` so you can do some more specific tests (like ensuring that 100% of the rewards have been distributed after the full time has elapsed)

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:247]: Can you be more specific here? We should ensure that 100% of the funds have been transferred from the vesting contract to the treasury after the clawback rather than just ensuring some amount has been moved. Also, can we use some more descriptive variables for `unclaimedaAmount` and `claimedAmount`. Maybe `initTreasuryBalance` and `finalTreasuryBalance`

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:211]: For every permissioned function you write, you want to add a test to ensure that it reverts if the wrong person calls it. Take a look at how its done using `context` here:
https://github.com/SetProtocol/index-coop-smart-contracts/blob/master/test/token/otcEscrow.spec.ts#L313

File: utils/deploys/deployToken.ts
Comment (ncitron) [inline utils/deploys/deployToken.ts:10]: Can you use the `../../typechain` formatting here  so that it falls in line with how the rest of the repo formats it (I agree that your way looks better though)

File: utils/deploys/deployToken.ts
Comment (ncitron) [inline utils/deploys/deployToken.ts:11]: Can you export FTCVesting in `utils/contracts.ts` so that you can import it with the rest of the contract types on line 4

**Comment by 0xModene** [comment]
@ncitron I think I have addressed all of your concerns

File: test/token/FTCVesting.spec.ts
Comment (ncitron) [inline test/token/FTCVesting.spec.ts:193]: Can we split this into two tests? One for a partial claim and one for a full claim. Typically you only want to call subject() once per test.

File: contracts/token/FTCVesting.sol
Comment (richardliang) [inline contracts/token/FTCVesting.sol:34]: What's the reasoning for removing this check in the Vesting contract?
`require(vestingBegin_ >= block.timestamp, "TreasuryVester.constructor: vesting begin too early");`

Is it retroactively vesting for contributors? Add a comment here noting why we removed the require for clarity.

**Review by richardliang — APPROVED** [review]
Checked against diff of Vesting.sol. Had a question on why the require for `vestingBegin_` but LGTM if that's resolved

File: contracts/token/FTCVesting.sol
Comment (0xModene) [reply contracts/token/FTCVesting.sol:34]: This was removed to allow retroactive vesting. We are currently approaching the cliff date on a few contributors, so this was removed to keep the accounting data straight with the hire date, vesting begin, etc.

**Pull request merged**
